matchzoo/contrib/models/esim.py

```diff
@@ -0,0 +1,238 @@
+"""ESIM model."""
+import keras
+import tensorflow as tf
+import keras.backend as K
+
+import matchzoo as mz
+from matchzoo.engine.param import Param
+from matchzoo.engine.base_model import BaseModel
+from matchzoo.engine.param_table import ParamTable
+
+
+class ESIM(BaseModel):
+    """
+    ESIM model.
+
+    Examples:
+        >>> model = ESIM()
+        >>> task = classification_task = mz.tasks.Classification(num_classes=2)
+        >>> model.params['task'] = task
+        >>> model.params['input_shapes'] = [(20, ), (40, )]
+        >>> model.params['lstm_dim'] = 300
+        >>> model.params['mlp_num_units'] = 300
+        >>> model.params['embedding_input_dim'] =  5000
+        >>> model.params['embedding_output_dim'] =  10
+        >>> model.params['embedding_trainable'] = False
+        >>> model.params['mlp_num_layers'] = 0
+        >>> model.params['mlp_num_fan_out'] = 300
+        >>> model.params['mlp_activation_func'] = 'tanh'
+        >>> model.params['mask_value'] = 0
+        >>> model.params['dropout_rate'] = 0.5
+        >>> model.params['optimizer'] = keras.optimizers.Adam(lr=4e-4)
+        >>> model.guess_and_fill_missing_params()
+        >>> model.build()
+    """
+
+    @classmethod
+    def get_default_params(cls) -> ParamTable:
+        """Get default parameters."""
+        params = super().get_default_params(with_embedding=True,
+                                            with_multi_layer_perceptron=True)
+
+        params.add(Param(
+            name='dropout_rate',
+            value=0.5,
+            desc="The dropout rate for all fully-connected layer"
+        ))
+
+        params.add(Param(
+            name='lstm_dim',
+            value=300,
+            desc="The dimension of LSTM layer."
+        ))
+
+        params.add(Param(
+            name='mask_value',
+            value=0,
+            desc="The value would be regarded as pad"
+        ))
+
+        return params
+
+    def _make_inputs(self) -> list:
```

Comment:
This is exactly the same as the base method, why overriding it?

Reply:
oh my bad, I was trying to pass mask initially. Forgot delete this after removing the mask input.

Reply:
> This is exactly the same as the base method, why overriding it?

oh my bad, I was trying to pass mask initially. Forgot delete this after removing the mask input.